fuel-core/src/model/fuel_block.rs

```diff
@@ -69,72 +67,84 @@ impl BlockHeight {
     }
 }
 
-#[derive(Clone, Debug, Default, Deserialize, Serialize, PartialEq)]
-pub struct TransactionCommitment {
-    pub sum: Word,
-    pub root: Bytes32,
-}
-
 #[derive(Clone, Debug, Deserialize, Serialize)]
-pub struct FuelBlockHeaders {
-    pub fuel_height: BlockHeight,
+pub struct FuelBlockHeader {
+    /// Fuel block height.
+    pub height: BlockHeight,
+    /// Ethereum block number.
+    pub number: BlockHeight,
+    /// Block header hash of the previous block.
+    pub prev_hash: Bytes32,
+    /// Merkle root of all previous block header hashes.
+    pub prev_root: Bytes32,
```

Comment:
Not exactly sure why is this needed. Shouldn't `prev_hash` be enought to connect the chain?

Reply:
The hash of the previous block header is sufficient, but if it's a MMR of all previous block header hashes then you can do log-sized proofs to past block headers rather than linear.

Reply:
I see, this is not something needed for client as we need to have all blocks in db for rebroadcasting, but for v2 contract?  But for even v2, wouldn't it be enought to have map of index->hash and there wouldn't be need MMR hashing and checking. Just Wondering, yes we have log-sized proof but where is it needed to have that reduction?

Reply:
Storing every block header hash would mean a storage write to a new slot for every block. If we use an MMR, then we only need to keep track of the tip's header hash, which is much cheaper.

Reply:
Which of our MT implementations supports MMR?

Reply:
The binary Merkle tree is actually an MMR 🤷

Reply:
I think I'll defer this to a separate task

Reply:
https://github.com/FuelLabs/fuel-core/issues/192